src/nodemetadata.cpp

```diff
@@ -200,6 +206,17 @@ void NodeMetadataList::clear()
 	m_data.clear();
 }
 
+int NodeMetadataList::countNonEmpty() const
```

Comment:
Can there be instead a function `hasNonEmpty`? That one can early return when it finds non empty entries.